SingleSource/UnitTests/Atomic/util.h

```diff
@@ -0,0 +1,66 @@
+//===--- util.h -- Utility functions shared by tests -------------- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _UTIL_H_
+#define _UTIL_H_
+
+#include <cstdint>
+#include <cstdlib>
+#include <iostream>
+
+#define TEST16 1
+
+static constexpr uint64_t V = 0x4243444546474849;
+static constexpr int kThreads = 10;
+static constexpr int kIterations = 1'000'000;
```

Comment:
A lot of other tests/benchmarks use the CMake variable `SMALL_PROBLEM_SIZE` to reduce the time it takes to run the test. Since you said these take a long time to run, could this variable be set to a smaller number so that it's still possible to run them on an emulator or slow FPGA? Another option could be to pass the number of iterations and threads as a command line argument which would make it easy to change the number of iterations dynamically.

Reply:
We can do that. FWIW, on x64 with AVX2, even 10^6 iterations is (rarely) flaky for aligned integer ALU operations.

Given that the speed of each operation affects contention rates affects flakiness of the test, it could make sense to have a different constant for each file. That is, since slow tests (like misaligned integers) can experience contention even with 10^5 iterations, we can speed up the test suite by setting the number of iterations to be inversely proportional to the speed of each operation.